### PR TITLE
feat: list & get doc versions

### PR DESCRIPTION
**Pull request by fontanierh**
- add new endpoint to list document versions
- allow retrieving a specific version of a doc

### CODE REVIEW

File: core/src/stores/sqlite.rs [hunk 1]
```diff
@@ -1005,7 +1005,13 @@ impl Store for SQLiteStore {
         project: &Project,
         data_source_id: &str,
         document_id: &str,
+        // NOTE: doesn't work on SQLITE
+        get_previous_version: bool,
     ) -> Result<Option<Document>> {
+        if get_previous_version {
+            return Err(anyhow!("get_previous_version not supported on SQLITE"));
```
Comment (fontanierh) [inline core/src/stores/sqlite.rs:1012]: Wondering if that's fine as we're deprecating ?

Comment (spolu) [reply core/src/stores/sqlite.rs:1012]: YES

**Review by spolu — COMMENTED** [review]
I think previous_version won't be enough for our use-case, my money on the fact that we'll need version as of timestamp?

**Comment by fontanierh** [comment]
Fair point

Are we fine relying on `created` for this ?

**Comment by spolu** [comment]
(this was before or after our discussion?)

**Comment by spolu** [comment]
Roger that

**Review by spolu — APPROVED** [review]
LGTM

**Review by Uzay-G — APPROVED** [review]

**Pull request merged**